data/summary/types/cgm.go

```diff
@@ -283,19 +321,33 @@ func (s *CGMStats) CalculateSummary() {
 	// currently only supports day precision
 	nextStopPoint := 0
 	nextOffsetStopPoint := 0
-	totalStats := &CGMBucketData{}
-	totalOffsetStats := &CGMBucketData{}
+	totalStats := &CGMTotalStats{}
+	totalOffsetStats := &CGMTotalStats{}
+	dayCounted := false
+	offsetDayCounted := false
 
 	for i := 0; i < len(s.Buckets); i++ {
 		currentIndex := len(s.Buckets) - 1 - i
 
+		// new day, reset day counting flag
+		if i%24 == 0 {
```

Comment:
24 is a magic number here.

While it's pretty obvious that it's the number of hours in a day, what's less obvious is that it's being used as a duration of each element in a bucket. 

That last bit is the important one, in the (unlikely) event that the duration should change, but even without that change, it helps any reader understand why 24 is appearing in the code.

Reply:
because this will soon change in many ways, I am voting to keep the magic number as it matches the implementation (no holes, always 24 hourly buckets, no variable duration), and will very soon be replaced by a more dynamic implementation which will outsource much of the time processing to db queries, removing the need for hourly offsets entirely.

Reply:
I'll trust your judgment then. :)